### ISSUE TITLE
[Feature] Add netty related metrics

### ISSUE THREAD

**Issue by wuchong** [alibaba/fluss#135]
<div type='discussions-op-text'>

<sup>Originally posted by **swuferhong** December  2, 2024</sup>
### Search before asking

- [X] I searched in the [issues](https://github.com/alibaba/fluss/issues) and found nothing similar.

### Motivation

Fluss uses `Netty` as the underlying network communication component, but it does not expose Netty-related metrics, such as the amount of network memory allocation and the number of requests etc. This makes it less convenient for optimizing network performance.

### Willingness to contribute

- [ ] I'm willing to submit a PR!</div>

**Comment by liyubin117** [comment]
I have digged into the feature and found that RequestsMetrics#createCoordinatorServerRequestMetrics is just a placeholder function, maybe we can enhance the above functions. I want to try this feature, Could you please assign this to me? thanks!
```
public static RequestsMetrics createCoordinatorServerRequestMetrics(
            MetricGroup serverMetricsGroup) {
        // now, no need to register any metrics of request for coordinator server
        return new RequestsMetrics(serverMetricsGroup, Collections.emptySet());
    }
```

**Comment by liyubin117** [comment]
@wuchong I am very interested in the feature, Could you please assign this to me, thanks!

**Comment by wuchong** [comment]
This issue was originally created by @swuferhong , have you been working on this issue? @swuferhong

**Comment by swuferhong** [comment]
> This issue was originally created by @swuferhong , have you been working on this issue? @swuferhong

No, I didn't do this. @liyubin117 You can take it. I've already assigned it to you.

**Comment by SteNicholas** [comment]
@liyubin117, any update? If no, I would like to introduce netty memory metrics.
